central/networkpolicies/datastore/datastore_sac_test.go

```diff
@@ -0,0 +1,396 @@
+	undostoremock "github.com/stackrox/rox/central/networkpolicies/datastore/internal/undostore/mocks"
+	"github.com/stackrox/rox/central/role/resources"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/bolthelper"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/fixtures"
+	"github.com/stackrox/rox/pkg/postgres/pgtest"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/testconsts"
+	"github.com/stackrox/rox/pkg/sac/testutils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"github.com/stretchr/testify/suite"
+	bolt "go.etcd.io/bbolt"
+)
+
+func TestNetworkPolicySAC(t *testing.T) {
+	suite.Run(t, new(networkPolicySACSuite))
+}
+
+type networkPolicySACSuite struct {
+	suite.Suite
+
+	datastore DataStore
+
+	pool *pgxpool.Pool
+
+	engine *bolt.DB
+
+	storage             store.Store
+	undoStore           undostore.UndoStore
+	undoDeploymentStore undodeploymentstore.UndoDeploymentStore
+
+	mockCtrl *gomock.Controller
+
+	testContexts         map[string]context.Context
+	testNetworkPolicyIDs []string
+}
+
+func (s *networkPolicySACSuite) SetupSuite() {
+	var err error
+	if features.PostgresDatastore.Enabled() {
+		ctx := context.Background()
+		src := pgtest.GetConnectionString(s.T())
+		cfg, err := pgxpool.ParseConfig(src)
+		s.NoError(err)
+		s.pool, err = pgxpool.ConnectConfig(ctx, cfg)
+		s.NoError(err)
+		pgdbStore.Destroy(ctx, s.pool)
+		s.storage = pgdbStore.New(ctx, s.pool)
+	} else {
+		s.engine, err = bolthelper.NewTemp(s.T().Name() + ".db")
+		s.NoError(err)
+		s.storage = boltStore.New(s.engine)
+	}
+	s.mockCtrl = gomock.NewController(s.T())
+	undomock := undostoremock.NewMockUndoStore(s.mockCtrl)
+	undodeploymentmock := undodeploymentstoremock.NewMockUndoDeploymentStore(s.mockCtrl)
+
+	s.datastore = New(s.storage, undomock, undodeploymentmock)
+
+	s.testContexts = testutils.GetNamespaceScopedTestContexts(context.Background(), s.T(), resources.NetworkPolicy)
+}
+
+func (s *networkPolicySACSuite) TearDownSuite() {
+	if features.PostgresDatastore.Enabled() {
+		s.pool.Close()
+	} else {
+		s.NoError(s.engine.Close())
+	}
+}
+
+func (s *networkPolicySACSuite) SetupTest() {
+	s.testNetworkPolicyIDs = make([]string, 0)
+}
+
+func (s *networkPolicySACSuite) TearDownTest() {
+	for _, id := range s.testNetworkPolicyIDs {
+		s.deleteNetworkPolicy(id)
+	}
+}
+
+func (s *networkPolicySACSuite) deleteNetworkPolicy(id string) {
+	s.NoError(s.datastore.RemoveNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], id))
+}
+
+func (s *networkPolicySACSuite) TestGetNetworkPolicy() {
+	networkPolicy := fixtures.GetScopedNetworkPolicy(uuid.NewV4().String(), testconsts.Cluster2, testconsts.NamespaceB)
+	err := s.datastore.UpsertNetworkPolicy(s.testContexts[testutils.UnrestrictedReadWriteCtx], networkPolicy)
+	s.NoError(err)
+	s.testNetworkPolicyIDs = append(s.testNetworkPolicyIDs, networkPolicy.GetId())
+	cases := map[string]struct {
```

Comment:
Probably should do this in a follow-up PR and not this one, but:
The case definitions are the same across all SAC integration tests we have, let's move them in a shared place similar to the generic search test cases we have under `pkg/sac/testutils`.

Reply:
There might be slight variations since some datastores check SAC at global resource level while others check SAC at target object level. But I agree it would be better to store the test case definitions in a central location.